### PR TITLE
Add subcommand version

### PR DESCRIPTION
**Pull request by muma378**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

To print verbose infromation about the client binary

```
$ ./bin/linux/amd64/kwokctl version
kwokctl: 8f82d7c
  BuildDate: 2023-09-14T06:24:39Z
  GoVersion: go1.20.7
  Compiler: gc
  Platform: linux/amd64
```

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: hack/releases.sh [hunk 1]
```diff
@@ -146,21 +146,34 @@ function main() {
...
-    LDFLAGS+=("-X sigs.k8s.io/kwok/pkg/consts.BinaryPrefix=${BINARY_PREFIX}")
+    LDFLAGS+=("-X ${PACKAGE}.BinaryPrefix=${BINARY_PREFIX}")
   fi
   if [[ "${BINARY_NAME}" != "" ]]; then
-    LDFLAGS+=("-X sigs.k8s.io/kwok/pkg/consts.BinaryName=${BINARY_NAME}")
+    LDFLAGS+=("-X ${PACKAGE}.BinaryName=${BINARY_NAME}")
+  fi
+  if [[ "${{GIT_COMMIT}}" != "" ]]; then
+    LDFLAGS+=("-x ${PACKAGE}.GitCommit=${{GIT_COMMIT}}")
+  fi
```
Comment (wzshiming) [inline hack/releases.sh:168]:
```suggestion
  if [[ "${GIT_COMMIT}" != "" ]]; then
    LDFLAGS+=("-x ${PACKAGE}.GitCommit=${GIT_COMMIT}")
  fi
```

File: pkg/consts/version.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+// default values when we build outside the Makefile context (e.g. go run, go build, or go test).
+var (
+	version      = "99.99.99"             // value from VERSION file
+	buildDate    = "1970-01-01T00:00:00Z" // output from `date -u +'%Y-%m-%dT%H:%M:%SZ'`
+	gitCommit    = ""                     // output from `git rev-parse HEAD`
+	gitTag       = ""                     // output from `git describe --exact-match --tags HEAD` (if clean tree state)
+	gitTreeState = ""                     // determined from `git status --porcelain`. either 'clean' or 'dirty'
+)
+
+// Version contains Argo version information
```
Comment (wzshiming) [inline pkg/consts/version.go:33]:
```suggestion
// Version contains Kwok version information
```

File: pkg/kwokctl/cmd/root.go [hunk 3]
```diff
@@ -42,7 +42,6 @@ func NewCommand(ctx context.Context) *cobra.Command {
 		Use:           "kwokctl [command]",
 		Short:         "Kwokctl is a Kwok cluster management tool",
 		Long:          "Kwokctl is a Kwok cluster management tool",
-		Version:       consts.Version,
```
Comment (wzshiming) [inline pkg/kwokctl/cmd/root.go:45]: Keep it

File: pkg/kwokctl/cmd/version/version.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2022 The Kubernetes Authors.
```
Comment (wzshiming) [inline pkg/kwokctl/cmd/version/version.go:2]:
```suggestion
Copyright 2023 The Kubernetes Authors.
```

File: hack/releases.sh (hunk 1 above)
Comment (muma378) [reply hack/releases.sh]: done

File: pkg/consts/version.go (hunk 2 above)
Comment (muma378) [reply pkg/consts/version.go]: done

File: pkg/kwokctl/cmd/root.go (hunk 3 above)
Comment (muma378) [reply pkg/kwokctl/cmd/root.go:45]: done

File: pkg/kwokctl/cmd/version/version.go (hunk 4 above)
Comment (muma378) [reply pkg/kwokctl/cmd/version/version.go]: done

File: pkg/consts/version.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import (
+	"fmt"
+	"runtime"
+)
+
+// Version information set by link flags during build. We fall back to these sane
+// default values when we build outside the Makefile context (e.g. go run, go build, or go test).
+var (
+	version   = "99.99.99"             // value from VERSION file
+	buildDate = "1970-01-01T00:00:00Z" // output from `date -u +'%Y-%m-%dT%H:%M:%SZ'`
```
Comment (wzshiming) [inline pkg/consts/version.go:29]:
```suggestion
	Version   = "99.99.99"             // value from VERSION file
	BuildDate = "1970-01-01T00:00:00Z" // output from `date -u +'%Y-%m-%dT%H:%M:%SZ'`
```

File: pkg/consts/version.go [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+	Compiler  string
+	Platform  string
+}
+
+func (v VersionInfo) String() string {
+	return v.Version
+}
+
+// GetVersion returns the version information
+func GetVersion() VersionInfo {
```
Comment (wzshiming) [inline pkg/consts/version.go:46]: Move to sigs.k8s.io/kwok/pkg/utils/version

Comment (muma378) [reply pkg/consts/version.go:46]: Do you want me to move VersionInfo to `sigs.k8s.io/kwok/pkg/utils/version` too?

File: pkg/consts/version.go (hunk 5 above)
Comment (muma378) [reply pkg/consts/version.go:29]: Version was already declared in the package, I'm not sure reusing the variable would not issue an error

Comment (wzshiming) [reply pkg/consts/version.go:29]: Yes, we just use the `Version`, why need to define a new `version`?

File: pkg/consts/version.go (hunk 6 above)
Comment (wzshiming) [reply pkg/consts/version.go:46]: Yes

**Pull request closed without merging**